synapse/storage/databases/main/stats.py

```diff
@@ -210,34 +210,56 @@ async def update_room_state(self, room_id: str, fields: Dict[str, Any]) -> None:
         * topic
         * avatar
         * canonical_alias
+        * guest_access
 
-        A is_federatable key can also be included with a boolean value.
+        An is_federatable key can also be included with a boolean value.
 
         Args:
             room_id: The room ID to update the state of.
             fields: The fields to update. This can include a partial list of the
                 above fields to only update some room information.
         """
-        # Ensure that the values to update are valid, they should be strings and
-        # not contain any null bytes.
+        # Ensure that the values to update are valid. str values should not
+        # contain any null bytes.
         #
         # Invalid data gets overwritten with null.
         #
         # Note that a missing value should not be overwritten (it keeps the
         # previous value).
-        sentinel = object()
-        for col in (
+        str_columns = [
             "join_rules",
             "history_visibility",
             "encryption",
             "name",
             "topic",
             "avatar",
             "canonical_alias",
-        ):
-            field = fields.get(col, sentinel)
-            if field is not sentinel and (not isinstance(field, str) or "\0" in field):
-                fields[col] = None
+            "guest_access",
+        ]
+        other_columns = [
+            "is_federatable",
+        ]
+
+        state_dict = {}
+        sentinel = object()
+
+        for column_name in fields.keys():
+            field = fields.get(column_name, sentinel)
+            if field is sentinel:
+                continue
+
+            if column_name in str_columns:
+                if not isinstance(field, str) or "\0" in field:
+                    # This field contained invalid str data, replace its contents with NULL
+                    state_dict[column_name] = None
+            elif column_name in other_columns:
+                if isinstance(field, str):
```

Comment:
This seems like a weird check? It means we can pass in an object and this wouldn't error?

Reply:
It's also not currently possible in the calling code for `is_federatable` to be anything other than a bool, even with malicious input. So we can probably just remove this check.

Reply:
That's the reasoning I didn't add one when previously poking at this!